File: Client/ecommerce-store/src/shared/models/product.ts
export interface IProduct {
  id: string;
  name: string;
  summary: string;
  description: string;
  imageFile: string;
  brands: IBrands;
  types: ITypes;
  price: number;
  priceAfterDiscount: number;
  hasDiscount: boolean;
  discountAmount: number;
  [key: string]: any;  // Add this line
}

export interface IBrands {
  id: string;
  name: string;
}

export interface ITypes {
  id: string;
  name: string;
}

File: Services/Catalog/Catalog.Core/Entities/Product.cs
using Discount.Core.Entities;
using MongoDB.Bson.Serialization.Attributes;

namespace Catalog.Core.Entities
{
     public class Product: BaseEntity
    {
        public string Name { get; set; }
        public string Description { get; set; }
        public string Summary { get; set; }
        public string ImageFile { get; set; }
        [BsonRepresentation(MongoDB.Bson.BsonType.Decimal128)]
        public decimal Price { get; set; }
        public decimal PriceAfterDiscount { get; set; }
        public ProductBrand Brands { get; set; }
        public ProductType Types { get; set; }
        public bool HasDiscount { get; set; } = false;
        public int DiscountAmount { get; set; } = 0;
        public Coupon? Coupon { get; set; } = null;
    }
}
